cpp/oneapi/dal/test/engine/csr_table_builder.hpp

```diff
@@ -0,0 +1,175 @@
+/*******************************************************************************
+* Copyright 2023 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/table/csr.hpp"
+#include "oneapi/dal/backend/primitives/utils.hpp"
+#include "oneapi/dal/test/engine/math.hpp"
+
+namespace oneapi::dal::test::engine {
+
+namespace pr = dal::backend::primitives;
+
+/**
+* Generates random CSR table based on inputs
+*/
+struct csr_table_builder {
+    using Float = float;
+    std::int64_t row_count_, column_count_;
+    float nonzero_fraction_;
+    sparse_indexing indexing_;
+    const dal::array<Float> data_;
+    const dal::array<std::int64_t> column_indices_;
+    const dal::array<std::int64_t> row_offsets_;
+
+    csr_table_builder(std::int64_t row_count,
+                      std::int64_t column_count,
+                      float nnz_fraction = 0.3,
```

Comment:
Not a practical fraction of non-zero elements.
Please decrease to 0.01 or less in order to generate the data that looks more similar with the real-world data.